docs/release-process.md

```diff
@@ -0,0 +1,128 @@
+  `HEAD` requirement)
+* Your *local* repository is cloned into `releases/$TAG`
+* `CHANGELOG.md` is checked to ensure it has an entry for `$TAG`
+* Distributables injected with `$TAG` are built
+* Tests are executed
+
+## Draft Phase
+### Push Version Tag Upstream
+
+First you must push your version tag upstream, so that an associated
+GitHub release may be created:
+
+    git push git@github.com:weaveworks/weave $TAG
+
+N.B. if you're testing the release process, push to your fork
+instead!
+
+### Create Draft Release
+
+You're now ready to draft your release notes:
+
+    bin/release draft [--pre-release]
+
+This has the following effects:
+
+* A [release](https://help.github.com/articles/about-releases) is
+  created in GitHub for `$TAG`. This release is in the draft state, so
+  it is only visible to contributors
+* The `weave` script is uploaded as an attachment to the release
+* If `--pre-release` is specified, the release will have the
+  pre-release attribute set (this affects the way GitHub displays the
+  release and modifies the behaviour of the publish phase)
+
+Navigate to https://github.com/weaveworks/weave/releases, 'Edit' the
+draft and input the release notes. When you are done make sure you
+'Save draft' (and not 'Publish release'!).
+
+Once the release notes have passed review, proceed to the publish
+phase.
+
+## Publish Phase
+### Move/Force Push `latest_release` Tag
+
+This step is only required for mainline (non pre-release) releases:
+
+    git tag -af -m "Release $TAG" latest_release $TAG
+    git push -f git@github.com:weaveworks/weave latest_release
+
+The `latest_release` tag *must* point at `$TAG`, *not* at `HEAD` -
+the build script will complain otherwise.
+
+N.B. if you're testing the release process, push to your fork
+instead!
+
+### Publish Release & Distributable Artefacts
+
+You can now publish the release and upload the remaining
+distributables to DockerHub:
+
+    bin/release publish
+
+This has the following effects:
+
+* Docker images are tagged `$TAG` and pushed to DockerHub
+* GitHub release moves from draft to published state
+
+Furthermore, if this is a mainline release (detected automatically
+from the GitHub release, you do not need to specify the flag again to
+the publish step)
+
+* Images tagged `latest` are updated on DockerHub
+* Release named `latest_release` is updated on GitHub
+
+
+## Troubleshooting
+
+There's a few things that can go wrong.
+
+ * If the build is wonky, e.g., the tests don't pass, you can delete
+   the directory in `./releases/`, fix whatever it is, move the
+   version tag (which should still be only local) and have another go.
+ * If the DockerHub pushes fail (which sadly seems to happen a lot),
+   you can just run `./bin/release publish` again.
+ * If it turns out you released the wrong thing, you can delete the
+   release on GitHub and start again; it's the only thing that has to
+   be manually deleted.
```

Comment:
Surely that is not correct. Images will have been published on dockerhub and tagged as 'latest', which means they will get picked up by the snaphot/master weave script. And the latest_release tag in gh will point at the just-released version, which means our published download URLs will pick that up.